training/src/anemoi/training/diagnostics/mlflow/azureml.py

```diff
@@ -0,0 +1,382 @@
+        log_hyperparams: bool | None = True,
+        on_resume_create_child: bool | None = True,
+        max_params_length: int | None = MAX_PARAMS_LENGTH,
+    ) -> None:
+        """Initialize the AnemoiMLflowLogger.
+
+        Parameters
+        ----------
+        aml_identity: str | None, optional
+            Type of authentication to fall back on for accessing the AzureML workspace.
+        aml_subscription_id: str | None, optional
+            The Azure subscription id
+        aml_resource_group: str | None, optional
+            Name of the Azure ML resource group
+        aml_workspace: str | None, optional
+            Name of the Azure ML workspace
+        azure_log_level: str, optional
+            Log level for all azure packages (azure-identity, azure-core, etc)
+        experiment_name : str, optional
+            Name of experiment, by default "lightning_logs"
+        project_name : str, optional
+            Name of the project, by default "anemoi"
+        run_name : str | None, optional
+            Name of run, by default None
+        tracking_uri : str | None, optional
+            Tracking URI of server, by default os.getenv("MLFLOW_TRACKING_URI")
+        save_dir : str | None, optional
+            Directory to save logs to, by default "./mlruns"
+        log_model : Literal[True, False, "all"], optional
+            Log model checkpoints to server (expensive), by default False
+        prefix : str, optional
+            Prefix for experiments, by default ""
+        resumed : bool | None, optional
+            Whether the run was resumed or not, by default False
+        forked : bool | None, optional
+            Whether the run was forked or not, by default False
+        run_id : str | None, optional
+            Run id of current run, by default None
+        fork_run_id : str | None, optional
+            Fork Run id from parent run, by default None
+        offline : bool | None, optional
+            Whether to run offline or not, by default False
+        authentication : bool | None, optional
+            Whether to authenticate with server or not, by default None
+        log_hyperparams : bool | None, optional
+            Whether to log hyperparameters, by default True
+        on_resume_create_child: bool | None, optional
+            Whether to create a child run when resuming a run, by default False
+        max_params_length: int | None, optional
+            Maximum number of params to be logged to Mlflow
+        """
+        import mlflow
+
+        self._resumed = resumed
+        self._forked = forked
+        self._flag_log_hparams = log_hyperparams
+        self._max_params_length = max_params_length
+
+        self._fork_run_server2server = None
+        self._parent_run_server2server = None
+        self._parent_dry_run = False
+
+        # <-- Azure specific stuff -->
+        # we don't need authenticate, this just lets us easily subclass the logger
```

Comment:
Regarding the authentication:
- above there is a mention to the `aml_identity` and that's used for azure authentication?
seems a bit confusing to have a entry in this logger with `authentication:None`. We could look at using the authentication flag already existing in the logger to carry out the azure one? (rather than SSO the AnemoiMlogger uses).